### PR TITLE
Bpk-2119 update minsdk

### PR DESCRIPTION
**Pull request by amolgupta**
This PR removes support for Android Kitkat. The `minSdk` required will now be `21`

- Moves resources under `v21` folders to default folders
- Removes Multidex support [see](https://developer.android.com/studio/build/multidex)
- Removes Vector Drawables support [see](https://developer.android.com/guide/topics/graphics/vector-drawable-resources)
- Updates CI and gulp files

### CODE REVIEW

File: Backpack/src/main/res/values/backpack.text.xml [hunk 1]
```diff
@@ -25,96 +25,115 @@
...
   <style name="bpkTextSm">
     <item name="android:fontFamily">sans-serif</item>
     <item name="android:textSize">14sp</item>
+    <item name="android:letterSpacing">0</item>
   </style>
 
   <style name="bpkTextXl">
     <item name="android:fontFamily">sans-serif</item>
     <item name="android:textSize">24sp</item>
+    <item name="android:letterSpacing">-0.024999999999999998</item>
```
Comment (antalindisguise) [inline Backpack/src/main/res/values/backpack.text.xml]: Where are these numbers documented as the correct ones?

File: app/build.gradle [hunk 2]
```diff
@@ -16,7 +16,6 @@ android {
     targetSdkVersion rootProject.ext.targetSdkVersion
     versionCode 1
     versionName "1.0"
-    multiDexEnabled true
     vectorDrawables.useSupportLibrary = true
```
Comment (antalindisguise) [inline app/build.gradle]: Can be dropped as well according to https://developer.android.com/guide/topics/graphics/vector-drawable-resources

File: templates/BackpackText.njk [hunk 3]
```diff
@@ -25,6 +25,7 @@
   <style name="{{text.name}}">
     <item name="android:fontFamily">{{text.fontFamily}}</item>
     <item name="android:textSize">{{text.size}}sp</item>
+    <item name="android:letterSpacing">{{text.letterSpacing}}</item>
```
Comment (antalindisguise) [inline templates/BackpackText.njk]: Shouldn't this go in a separate PR?

File: Backpack/src/main/res/values/backpack.text.xml (hunk 1 above)
Comment (amolgupta) [reply Backpack/src/main/res/values/backpack.text.xml]: those are type scale changes and are generated from tokens. I can confirm they are correct.

File: app/build.gradle (hunk 2 above)
Comment (amolgupta) [reply app/build.gradle]: good catch. I removed it in the library already, removing from the demo app now.

File: templates/BackpackText.njk (hunk 3 above)
Comment (amolgupta) [reply templates/BackpackText.njk]: This is the part that generates the type scale values. CI would fail if not done along with this PR.
